### DISCUSSION TITLE
osu is glitching on xwayland

### DISCUSSION THREAD

**Discussion by anispwyn** [ppy/osu#29148] [Q&A] [answered]
i'm currently using hyprland and osu is running under xwayland (using iGPU since for some reason, using dgpu (nvidia rtx 3050ti giving lower frame per seconds)

and for some reason, osu started glitching, this seems to happen on any page that has list (including multi and daily)

video: https://streamable.com/8h52fe

logs: 
[compressed-logs.zip](https://github.com/user-attachments/files/16401449/compressed-logs.zip)

using aur/osu-lazer-bin

**Comment by Stoppedpuma** [comment, marked as answer]
This could be for two large reasons or even a mixture of both. Since you seem to be on arch if you're using `hyprland-git`, hyprland has moved to aquamarine from wlroots. This currently has a lot of issues and shouldn't have been merged in it's current state in my opinion. The second reason is the fact that you're using an NVIDIA GPU, NVIDIA has a lot of problems on Wayland. If you are on hyprland-git I would recommend installing the regular Hyprland package for now and see if that makes a difference.

**Comment by anispwyn** [comment]
It is hyprland issue, thank you!
